backend/tarsy/integrations/mcp/client.py

```
@@ -76,11 +76,26 @@ async def initialize(self):
-                logger.info(f"Successfully initialized MCP server: {server_id}")
+                # Create and initialize session using shared helper with timeout
+                # Use a reasonable timeout to prevent hanging during startup
+                try:
+                    session = await asyncio.wait_for(
+                        self._create_session(server_id, server_config),
+                        timeout=30.0  # 30 second timeout for initialization
+                    )
+                    self.sessions[server_id] = session
+                    logger.info(f"Successfully initialized MCP server: {server_id}")
+                except asyncio.TimeoutError:
+                    raise Exception(f"Server initialization timed out after 30 seconds") from None
```

Ruff (0.14.1)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)